pkg/proxy/envoyproxy.go

```diff
@@ -45,9 +47,46 @@ func initEnvoy(runDir string, xdsServer *envoy.XDSServer, wg *completion.WaitGro
 				xdsServer.AddMetricsListener(uint16(option.Config.ProxyPrometheusPort), wg)
 			}
 		}
+
+		if envoyAdminClient != nil && !option.Config.DisableEnvoyVersionCheck {
+			if err := checkEnvoyVersion(); err != nil {
+				log.WithError(err).Error("Envoy: Version check failed")
+			}
+		}
 	})
 }
 
+func checkEnvoyVersion() error {
+	const versionRetryAttempts = 20
+	const versionRetryWait = 500 * time.Millisecond
```

Comment:
is there no way of making this event based instead of waiting for a specific timeout? I'm worried that this will (if even enabled) cause flakes in CI due to dependence on timing

Reply:
@bimmlerd 

the version check feature is enabled by default - so we would see flakes if this becomes a problem.

changing it to something "event based" would only help in case the flakyness would occur within the retry wait duration - which is quite low with 500ms.

beside this, the whole version check is subject of discussion because we kind of want to support a broader range of envoy versions in the future.

Reply:
actually, we don't error out - we're only writing an error message to the log. and in most of the cases envoy starts faster than the envoy (within ~1s) - only the embedded case where the check gets executed immediately after the start of envoy might need 1, 2 retries.

Reply:
:facepalm: I apparently couldn't read two lines up

alright, fine by me